Recast/PlayerViewController.swift

```diff
@@ -8,59 +8,344 @@
 
 import UIKit
 import SnapKit
+import AVFoundation
+
+private var playerViewControllerKVOContext = 0
 
 class PlayerViewController: UIViewController {
 
     // MARK: - Variables
+    @objc let player = AVQueuePlayer()
+
+    var currentTime: Double {
+        get {
+            return CMTimeGetSeconds(player.currentTime())
+        }
+        set {
+            let newTime = CMTimeMakeWithSeconds(newValue, preferredTimescale: 1)
+            player.seek(to: newTime, toleranceBefore: CMTime.zero, toleranceAfter: CMTime.zero)
+        }
+    }
+
+    var duration: Double {
+        guard let currentItem = player.currentItem else { return 0.0 }
+        return CMTimeGetSeconds(currentItem.duration)
+    }
+
+    var rate: Float {
+        get {
+            return player.rate
+        }
+        set {
+            player.rate = newValue
+        }
+    }
+
+    var playerLayer: AVPlayerLayer? {
+        return playerView.playerLayer
+    }
+
+    /*
+     A formatter for individual date components used to provide an appropriate
+     value for the `startTimeLabel` and `durationLabel`.
+     */
+    let timeRemainingFormatter: DateComponentsFormatter = {
```

Comment:
should we put this in a utilities file? i used a dateformatter for the episode cells that we can probs move over too